SCAutolib/models/gui.py

```diff
@@ -377,7 +379,18 @@ def click_on(self, key: str, timeout: float = 30):
     def kb_write(self, *args, **kwargs):
         # delay is a workaround needed for keyboard library
         kwargs.setdefault('delay', 0.1)
-        keyboard.write(*args, **kwargs)
+
+        word = args[0]
+        last = ""
+        for char in word:
+            if char.isupper():
+                if last != "":
+                    keyboard.write(*[last], **kwargs)
+                    last = ""
+                keyboard.send(f"shift+{char.lower()}")
+            else:
+                last = f"{last}{char}"
+        keyboard.write(*[last], **kwargs)
```

Comment:
Don't really get the point of this change. Will that work if you send "shift+char"? Doesn't it need a numerical code for the shift to be emulated? I think we have something like that with ctrl.

Reply:
This was a hack to handle uppercase characters.   The keyboard.write() can only handle lowercase apparently.   With send() though you can do key combos like shift+char.   So far it seems that it handles it appropriately.   I tested with a manual password login with mixed  case chars and numbers and it passed the key presses appropriately and logged in.

Reply:
Ok sounds fine to me!